kubelet/tests/test_kubelet.py

```diff
@@ -525,6 +535,22 @@ def test_report_container_state_metrics(monkeypatch, tagger):
         raise AssertionError('kubernetes.containers.state.* was submitted without a reason')
 
 
+def test_no_tags_no_metrics(monkeypatch, aggregator, tagger):
+    # Reset tagger without tags
+    tagger.reset()
+    tagger.set_tags({})
+
+    check = mock_kubelet_check(monkeypatch, [{}])
+    check.check({"cadvisor_metrics_endpoint": "http://dummy/metrics/cadvisor", "kubelet_metrics_endpoint": ""})
+
+    # Test that we get only the node related metrics (no calls to the tagger for these ones)
+    metric1 = ensure_unicode('kubernetes.memory.capacity')
+    metric2 = ensure_unicode('kubernetes.cpu.capacity')
+    assert (
+        metric1 in aggregator.metric_names and metric2 in aggregator.metric_names and len(aggregator.metric_names) == 2
+    )
```

Comment:
You can/should use this instead:

```
aggregator.assert_metric('kubernetes.memory.capacity')
aggregator.assert_metric('kubernetes.cpu.capacity')
aggregator.assert_all_metrics_covered()
```

If needed, you can also assert the count:


```
aggregator.assert_metric('kubernetes.memory.capacity', count=1)
aggregator.assert_metric('kubernetes.cpu.capacity', count=1)
aggregator.assert_all_metrics_covered()
```

Reply:
I want to assert that all metrics are not covered but 'kubernetes.memory.capacity' and 'kubernetes.cpu.capacity', because of my change. For this test I remove the tags so it should not send any metrics beside these two.
I did not know how to do this.

Reply:
My bad, I just tested it and it works. I just did not understand what it did! 👍 
Comitting the change